File: workflow/scripts/plot_horsepower_vs_mpg.R
library(readr)
library(ggplot2)

cars <- read_tsv(snakemake@input[[1]], show_col_types = FALSE)
svg(snakemake@output[[1]])
ggplot(cars, aes(`miles per gallon`, horsepower)) +
  geom_point() +
  theme_classic(16)
dev.off()

File: workflow/scripts/plot_horsepower_vs_mpg.py
import altair as alt
import polars as pl
alt.data_transformers.enable("vegafusion")

data = pl.read_csv(snakemake.input[0], separator="\t")

chart = alt.Chart(data).mark_point(tooltip=True).encode(
    alt.X("miles per gallon"),
    alt.Y("horsepower"),
    alt.Color("origin").scale(scheme="accent"),
).interactive()

chart.save(snakemake.output[0])
